packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -104,6 +195,11 @@ export class ModelConnectionTransformer extends Transformer {
             throw new InvalidDirectiveError(`Object type ${relatedTypeName} must be annotated with @model.`)
         }
 
+        if (getDirectiveArgument(directive)("fields")) {
+            this.newParameterization(parent, field, directive, ctx)
```

Comment:
Nice clean way of diving into the new logic. Can you please place a comment here explaining why and that the behavior is changing based on the fields presence.